### PR TITLE
add job name search to translation progress (WP-611)

### PR DESCRIPTION
**Pull request by vsolovei-smartling**

### CODE REVIEW

File: inc/Smartling/WP/Table/SubmissionTableWidget.php [hunk 1]
```diff
@@ -280,7 +281,7 @@ public function prepare_items(): void
             $searchBlock = ConditionBlock::getConditionBlock(ConditionBuilder::CONDITION_BLOCK_LEVEL_OPERATOR_OR);
 
             $searchFields = [
-                SubmissionEntity::FIELD_SOURCE_TITLE,
+                JobEntity::FIELD_JOB_NAME,
```
Comment (PavelLoparev) [inline inc/Smartling/WP/Table/SubmissionTableWidget.php:284]: Didn't get it, we replaced the filter by source title with filter by job name?

Comment (vsolovei-smartling) [reply inc/Smartling/WP/Table/SubmissionTableWidget.php:284]: SubmissionEntity::FIELD_SOURCE_TITLE was present twice in the array, I've replaced one occurence with Job Name

**Review by PavelLoparev — CHANGES_REQUESTED** [review]
`QueryBuilder` is used not only in submission s manager class. Could we also cover with tests existing queries generated by query builder?
- In `Queue.php`
- In `MultilingualPressConnector.php`
- In `AbsoluteLinkedAttachmentCoreHelper.php`
- In `JobAbstract.php`
- In `SettingsManager.php`
- In `AcfOptionEntity.php`

File: tests/Smartling/Submissions/SubmissionManagerTest.php [hunk 2]
```diff
@@ -42,7 +42,7 @@ public function testSearch_EmptyConditionBlock()
     public function testSearch_ConditionBlockWithConditions()
     {
         $db = clone $this->db;
-        $db->expects($this->once())->method('fetch')->with('SELECT COUNT(*) AS `cnt` FROM `wp_posts` WHERE ( ( `post_status` = \'draft\' ) )')->willReturn([$this->result]);
+        $db->expects($this->once())->method('fetch')->with("SELECT COUNT(DISTINCT s.id) AS cnt\n     FROM wp_smartling_submissions AS s\n        LEFT JOIN wp_smartling_submissions_jobs AS sj ON s.id = sj.submission_id\n        LEFT JOIN wp_smartling_jobs AS j ON sj.job_id = j.id  WHERE ( ( post_status = 'draft' ) )")->willReturn([$this->result]);
```
[the remark this answers is not in the document]
Comment (vsolovei-smartling) [reply tests/Smartling/Submissions/SubmissionManagerTest.php:47]: The manager only uses whatever conditions were provided. Changed to reduce confusion in  103146d2f89df7c081a6488df38d0f45cb28803e

[the remark this answers is not in the document]
Comment (vsolovei-smartling) [reply tests/Smartling/Submissions/SubmissionManagerTest.php:57]: Done

**Review by PavelLoparev — APPROVED** [review]
I'm against this approach with manual sql query generation. But since it's already used in the project it seems like refactoring is out of this task's scope.

Did you try to run these new tests on current master branch? I just want to make sure that existing queries are not affected and only one query is changed.

**Comment by vsolovei-smartling** [comment]
Yes, I got that intent and ran them on master. Only failing tests were for SubmissionManager, but this was expected, as they are changed in branch.

**Comment by PavelLoparev** [comment]
Ok, great

**Pull request merged**